### PR TITLE
Sign up form pull request

### PR DESCRIPTION
**Pull request by curesh**
Implement basic sign up form that updates user table on airtable. Still need to integrate airlock.

### CODE REVIEW

File: src/components/SignUp/SignUpForm.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+const base = new Airtable({ apiKey: airtableConfig.apiKey }).base(airtableConfig.baseKey);
+
+const INITIAL_FORM_STATE = {
+  username: '',
+  fullname: '',
+  password: '',
+};
+
+// Main fields are, username (email), password, First Name, Last Name, Role
```
Comment (ryanmtran) [inline src/components/SignUp/SignUpForm.js]: For sake of standardization, make sure to remove comments unless there is a really ambiguous function or set of lines that require explicit descriptions

File: src/components/SignUp/SignUpForm.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    event.preventDefault();
+    base('Users').create([
+      {
+        fields: {
+          username: formState.username,
+          password: formState.password,
+          'user type': role,
+          'display name': formState.fullname,
+          approval: 'unapproved',
+          'seller quotes': [
```
Comment (ryanmtran) [inline src/components/SignUp/SignUpForm.js]: We haven't implemented quotes yet so make sure to remove lines related to quotes

Comment (ryanmtran) [inline src/components/SignUp/SignUpForm.js]: Make sure to keep the fields like username, password, user type, etc consistent, so use single quotes for all

Comment (curesh) [reply src/components/SignUp/SignUpForm.js]: Eslint throws an error for putting quotes around approval:
"Unnecessarily quoted property 'approval' found.eslint(quote-props)"

Should I figure out a way to circumvent the error?

**Review by ryanmtran — CHANGES_REQUESTED** [review]
Great job! See comments for minor changes. Moving forward, try to implement the following features:

1. When any of the text inputs are empty or the selector is at its blank default state, disable the button
2. When the user clicks submit and the user account is being created/registered, disable all text inputs, the select, and submit button during the load time. This might not be relevant/testable now, but it should definitely be translatable to your Login Page.
3. Implement a Re-enter Password text input, and make sure the contents of this text input matches your password text input during submit. If it does not, create an alert to the user that these two inputs don't match. 

Here's the SignIn and SignUp forms from PTV for inspiration
https://github.com/lablueprint/ptv-web-console/blob/master/src/components/SignIn/SignInForm.js

https://github.com/lablueprint/ptv-web-console/blob/master/src/components/SignUp/SignUpForm.js

File: src/components/SignUp/SignUpForm.js (hunk 2 above)
Comment (ryanmtran) [reply src/components/SignUp/SignUpForm.js]: In that case let's refactor the names on Airtable go from user type to user-type and we'll just change that accordingly throughout our code. So change user type -> user-type, display type -> display-type, etc.

File: src/components/SignUp/SignUpForm.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+import './SignUp.css';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import Airlock from '@calblueprint/airlock';
+
+Airlock.configure({
+  endpointUrl: 'http://localhost:3000',
+  apiKey: 'airlock',
+});
+const base = Airlock.base('appBsavky1VJ0HK23');
```
Comment (ryanmtran) [inline src/components/SignUp/SignUpForm.js]: Make sure you refer to the base from the .env file
const airtableConfig = {
  baseKey: process.env.REACT_APP_AIRTABLE_BASE_KEY,
};

**Review by ryanmtran — COMMENTED** [review]
Just one nitpick about the base api key! Thank you so much Chandra I'm so proud! This is an AMAZING first feature and I can't wait to see what else we create!

**Pull request closed without merging**